sigmastate/src/test/scala/sigmastate/utxo/DistributedSigSpecification.scala

```diff
@@ -219,22 +213,22 @@ class DistributedSigSpecification extends SigmaTestingCommons {
     val prop = compile(env, """atLeast(3, Coll(pubkeyA, pubkeyB, pubkeyC, pubkeyD))""").asSigmaProp
 
     // Alice, Bob and Carol are signing
-    val (rBob, aBob) = DiffieHellmanTupleInteractiveProver.firstMessage(pubkeyBob)
-    val dlBKnown: Hint = RealCommitment(pubkeyBob, aBob)
+    val bobHints = proverB.generateCommitments(prop.treeWithSegregation, ctx)
+    val dlBKnown: Hint = bobHints.realCommitments.head
 
-    val (rCarol, aCarol) = DiffieHellmanTupleInteractiveProver.firstMessage(pubkeyCarol)
-    val dlCKnown: Hint = RealCommitment(pubkeyCarol, aCarol)
+    val carolHints = proverC.generateCommitments(prop.treeWithSegregation, ctx)
+    val dlCKnown: Hint = carolHints.realCommitments.head
 
     val bagA = HintsBag(Seq(dlBKnown, dlCKnown))
     val proofAlice = proverA.prove(prop, ctx, fakeMessage, bagA).get
 
     val bagC = proverC.bagForMultisig(ctx, prop, proofAlice.proof, Seq(pubkeyAlice), Seq(pubkeyDave)) ++
-      HintsBag(Seq(dlBKnown, OwnCommitment(pubkeyCarol, rCarol, aCarol)))
+      HintsBag(Seq(dlBKnown, carolHints.ownCommitments.head))
     val proofCarol = proverC.prove(prop, ctx, fakeMessage, bagC).get
 
     val bagB = (proverB.bagForMultisig(ctx, prop, proofAlice.proof, Seq(pubkeyAlice), Seq(pubkeyDave)) ++
                 proverB.bagForMultisig(ctx, prop, proofCarol.proof, Seq(pubkeyCarol)))
-                  .addHint(OwnCommitment(pubkeyBob, rBob, aBob))
+                  .addHint(bobHints.ownCommitments.head)
```

Comment:
If I got it right, Carol's proof should also fail verification? Please add the check.

Reply:
done